core/string/ustring.cpp

```diff
@@ -3761,6 +3766,16 @@ String String::strip_escapes() const {
 	return new_string;
 }
 
+String String::strip_bbcode() const {
+#ifdef MODULE_REGEX_ENABLED
+	RegEx regex("\\[.*?\\]");
+	return regex.sub(*this, "", true);
+#else
+	WARN_PRINT("Stripping BBCode requires the regex module to be enabled. Returning the original string as-is.");
+	return *this;
+#endif
+}
```

Comment:
```suggestion
String String::strip_bbcode() const {
	String result;
	int from = 0;
	while (true) {
		int lb_pos = find_char('[', from);
		if (lb_pos < 0) {
			break;
		}
		int rb_pos = find_char(']', lb_pos + 1);
		if (rb_pos < 0) {
			break;
		}
		result += substr(from, lb_pos - from);
		from = rb_pos + 1;
	}
	result += substr(from);
	// Remove remaining special characters to avoid security issues with concatenation.
	return result.replace("[", "").replace("]", "");
}
```

Reply:
I've updated the documentation accordingly to mention the behavior change (extraneous brackets are now removed).